src/TestContainers/Docker.hs

```diff
@@ -580,35 +602,43 @@ run request = do
         Just . (prefix <>) . ("-" <>) . pack
           <$> replicateM 6 (Random.randomRIO ('a', 'z'))
 
-  let dockerRun :: [Text]
-      dockerRun =
+  -- Instead of using `docker run`, we use the more manual `docker create` + `docker start`.
+  -- This allows to get the container ID early from `docker create`, and thus
+  -- optionally copy files using `docker cp`.
+  let dockerCreate :: [Text]
+      dockerCreate =
         concat $
-          [["run"]]
-            ++ [["--detach"]]
```

Comment:
Note that the `--detach` flag isn't supported by `docker start`, because this is the default behavior